components/suggest/src/weather.rs

```diff
@@ -89,10 +89,19 @@ impl SuggestDao<'_> {
 
         let max_chunk_size =
             std::cmp::max(g_cache.max_name_word_count, w_cache.max_keyword_word_count);
-        let kw = query.keyword.to_lowercase();
+
+        // Lowercase, strip punctuation, and split the query into words.
+        let kw_lower = query.keyword.to_lowercase();
```

Comment:
Nit: this seems slightly easier to read to me.

```rs
    let normalized = query
        .keyword
        .to_lowercase()
        .replace(|c| c.is_ascii_punctuation(), "");
    let words: Vec<_> = normalized.split_whitespace().collect();
```

Reply:
Thanks!

This doesn't quite work because we want (IMO) to split on punctuation as well as spaces, not collapse it. e.g., "rochester,ny" should be split into "rochester" and "ny". It does work if we do `replace(|c: char| c.is_ascii_punctuation(), " ")`, so no big deal.

I do agree my version is a little harder to understand, but it has the benefit of not allocating a new string, so I think I'd prefer to leave it if that's OK.

Reply:
Ah right, I guess we could replace with space, but I am fine to keep yours as well.